js/node.js

```diff
@@ -0,0 +1,212 @@
+
+                const link_elem = document.createElement("a");
+
+                let name;
+                let description;
+
+                if (link.rel === "service") {
+                    name = service.name;
+                } else if (link.rel === "service-desc" ) {
+                    name = `${service.name} description`;
+                    description = service.description;
+                }else if (link.rel === "service-doc") {
+                    name = `${service.name} documentation`;
+                }else {
+                    name = `${service.name} ${link.rel}`;
+                }
+
+                Object.entries(link).forEach(([attr, value]) => link_elem.setAttribute(attr, value))
+                link_elem.innerText = name;
+
+                //Fill in columns with links and information for each service, service documentation
+                // and service description
+                if(link_elem.rel === "service"){
+                    table_row.appendChild(table_cell_service);
+                    table_cell_service.appendChild(link_elem);
+                }else if(link_elem.rel === "service-doc"){
+                    table_row.appendChild(table_cell_documentation);
+                    table_cell_documentation.appendChild(link_elem);
+                }else if(link_elem.rel === "service-desc"){
+                    const descriptionTextNode = document.createTextNode(description);
+                    table_row.appendChild(table_cell_description);
+                    table_cell_description.appendChild(descriptionTextNode);
+                    table_cell_description.appendChild(document.createElement("br"))
+                    table_cell_description.appendChild(link_elem);
+                }else{
+                    table_row.appendChild(table_cell_other);
+                    table_cell_other.appendChild(link_elem)
+                }
+            })
+        })
+        return services_table;
+    }
+}
+
+document.addEventListener("DOMContentLoaded", function () {
+    const githubURL = "https://raw.githubusercontent.com/DACCS-Climate/DACCS-node-registry/current-registry/node_registry.json";
+    const url_params = new URLSearchParams(window.location.search)
+    const node_name = url_params.get("node");
+    fetch(githubURL).then(resp => resp.json()).then(json => {
+        const node_info = json[node_name];
+        if (typeof node_info === "undefined") {
+            if (url_params.has("node") && node_name.length > 0) {
+                window.alert(`Error: no node named ${node_name}`);
+            } else {
+                window.alert("Error: no node name specified");
+            }
+        } else {
+            Object.entries(node_info).forEach(([key, val]) => {
+                const elem = document.getElementById(key);
+
+                if (elem !== null) {
+
+                    switch(key){
+                        case "affiliation":
```

Comment:
All of these case statements make the converters hash completely redundant.
The point of the converts hash is to keep the specific conversion code for each key in one place so you _don't_ have to do something like this in this function.
Please review this code again and either move the logic into the converters table or if you have some objection to that strategy let's discuss it please.

Reply:
Case statements removed.  Moved logic to hash.